### ISSUE TITLE
Only GitHub?

### ISSUE THREAD

**Issue by jimrandell** [stateful/runme-web-extension#10]
Hello,
first, thank you for your efforts, i think runme is amazing!
...and while it great in vscode and github, i have not been able to figure out (it its at all possable) how to use this on code in any site other than github (my particular interest is using it in gitlab).

so, this is likely a feature request, but if i'm lucky, it will be a bug or support issue, and someone will have a solution.

Thanks Again,
-Jim

**Comment by christian-bromann** [comment]
@jimrandell thanks for raising an issue.

I will look into bringing this into Gitlab this week. Do you have any other feedback on Runme?

**Comment by jimrandell** [comment]
Thank you for your inquisitive response, its encouraging to know someone's interested in my opinion!

So, very specifically, I would like to have runme execute code in codeblocks in gitlab and bookstak (wiki).

But in the bigger picture, i have two things i would love runme to do (one of which is about marquee, but i guess it's really about runme, depending on how you implement it): 
 - Add a button to any codeblock in any browser page to execute the code. 
 - Run an embedded widget (iframe) in a marquee dashboard widget.
   - this marquee widget would allow you to provide, and will then render:
     - HTML code
     - A reference to a notebook cell containing html
   - (alternatively: a sql query, the results of which would display as a table when the results return multiple rows)

On an unrelated note, i see your in SF. how is it these days?
i lived for 20 years at Dubose & 14th (around the corner from Zeitgeist). I left 5 years ago to take a job in the UK. i very much miss the San francisco I moved to in '96! 
...but why think of May in November, when December is all that you'll get.

**Comment by christian-bromann** [comment]
> * Add a button to any codeblock in any browser page to execute the code.

Can you explain a little bit what you mean by that? We can't run e.g. shell code in the browser so we need to connect with a Runme instance that is running on a machine somewhere. So let's say the extensions allows you to connect with a remote machine. Then we could have the code cell be executed on that machine and report back in the browser. Is that something that would be interesting?

> * Run an embedded widget (iframe) in a marquee dashboard widget.

I am not sure exactly if I understand both of your ideas. We have a Runme widget in Marquee which allows you to select between all available notebooks in your workspace and have it open for you. What would like to see enhanced for this workflow or do you have a different in mind.

> On an unrelated note, i see your in SF. how is it these days?
> i lived for 20 years at Dubose & 14th (around the corner from Zeitgeist). I left 5 years ago to take a job in the UK. i very much miss the San francisco I moved to in '96!

Oh awesome, well there are good and bad developments but overall I enjoy the city a lot. Just moved here last September. Love Zeitgeist, they have good German beer 🍻

**Comment by jimrandell** [comment]
> > * Add a button to any codeblock in any browser page to execute the code.
> 
> Can you explain a little bit what you mean by that?

Yes, but I'm a bit overwhelmed today, and I want to let think about how to concisely express my thoughts on this


> * Run an embedded widget (iframe) in a marquee dashboard widget.

Perhaps this functionality already exists, but I haven't thought it through:
(btw, this is a workaround for a more specific use:)
  - I would like to use widgets from services that allow their widgets to be embedded within HTML, by providing the user with code to 'embed'. 
    - this usually takes the form of an iframe.
    - example from Metabase running on a local instance, which i would like to use on my marquee dashboard:
      - `<iframe
    src="http://localhost:3000/public/dashboard/7d82ea84-8c8c-47eb-8106-ff0328e4a61e"
    frameborder="0"
    width="800"
    height="600"
    allowtransparency></iframe>`
      - (the rendered widget)
      - ![image](https://github.com/stateful/runme-web-extension/assets/3543506/e1a6051b-c596-4ae8-882b-74554b164062)


  - as I said, this is a workaround for a use that I assume is (currently) outside the scope.
    - I want a dashboard that displays the results of SQL queries
      - I currently use Metabase for this, which provides this functionality with brilliant simplicity
      - **accessing my metabase widgets in marqees would work fine, if its possible!!!!**


> Well there are good and bad developments but overall.

(i know I'm going to sound like the old guys I used to overhear chatting in bars when I moved there in '96, but)
The city wasn't the same city I moved to 20 years prior (but to be fair, I wasn't really the same man)! 
the SF I moved to had a surplus of apartments available starting at around $650 for a 1 bedroom. Because I had a dog, I took a place a bit more than I could afford at the time, 900 for a large two-bedroom on 14th and Valencia (considered a very bad neighborhood at the time...friends told me i was paying way too much). When I moved out in 2016, I was paying $1250 (because they can't raise your rent more than the annual rate of inflation), I learned from a neighbor that they rented it out for $5,400!
